File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let inputLines = [];
rl.on("line", (line) => {
  inputLines.push(line);
});

rl.on("close", () => {
  const [N, K] = inputLines[0].split(" ").map(Number);
  const ans = [];
  const arr = Array.from({ length: N }, (_, i) => i + 1);
  let num = 0;

  for (let i = 0; i < N; i++) {
    num += K - 1;
    if (num >= arr.length) {
      num %= arr.length;
    }
    ans.push(arr.splice(num, 1)[0].toString());
  }

  console.log("<" + ans.join(", ") + ">");
});

File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.py
import sys
from collections import deque

input = sys.stdin.readline()

N, K = map(int, (input.split()))
queue = deque(list(range(1, N + 1)))

ans = []

temp = 1
while len(queue) >= 1:
    if temp != K:
        queue.append(queue.popleft())
        temp += 1
    else:
        ans.append(queue.popleft())
        temp = 1

res = ans + list(queue)

print("<" + ", ".join(map(str, res)) + ">")
